Engine/Source/Runtime/Rendering/Resources/ShaderResource.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "Base/Template.h"
+#include "IResource.h"
+#include "Rendering/ShaderType.h"
+
+#include <string>
+#include <vector>
+
+namespace engine
+{
+
+// TODO : Template?
+class ShaderResource : public IResource
```

Comment:
Template can use enum as input type to specialize two program types about cs only, vs only, vs + fs three types. I think it is OK to have different types of ShaderResource. We can manage it by IResource interfaces.